modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/ObjectEntryOpenAPIContributor.java

```diff
@@ -121,26 +122,34 @@ else if (key.contains("objectRelationshipName")) {
 
 					ObjectDefinition relatedObjectDefinition = entry.getValue();
 
-					String relatedSchemaName = getSchemaName(
-						relatedObjectDefinition);
+					String relatedSchemaName = null;
 
-					if (_addRelatedSchemas) {
-						_addObjectRelationshipSchema(
-							relatedObjectDefinition, openAPI,
-							relatedSchemaName);
-					}
+					if (!relatedObjectDefinition.isUnmodifiableSystemObject() ||
+						FeatureFlagManagerUtil.isEnabled("LPS-162966")) {
+
+						relatedSchemaName = getSchemaName(
+							relatedObjectDefinition);
 
-					if (Objects.equals(
-							objectRelationship.getType(),
-							ObjectRelationshipConstants.TYPE_MANY_TO_MANY) ||
-						(Objects.equals(
-							objectRelationship.getType(),
-							ObjectRelationshipConstants.TYPE_ONE_TO_MANY) &&
-						 (objectRelationship.getObjectDefinitionId1() ==
-							 _objectDefinition.getObjectDefinitionId()))) {
-
-						_addObjectRelationshipPathItem(
-							key, objectRelationship, paths, relatedSchemaName);
+						if (_addRelatedSchemas) {
+							_addObjectRelationshipSchema(
+								relatedObjectDefinition, openAPI,
+								relatedSchemaName);
+						}
+
+						if (Objects.equals(
+								objectRelationship.getType(),
+								ObjectRelationshipConstants.
+									TYPE_MANY_TO_MANY) ||
+							(Objects.equals(
+								objectRelationship.getType(),
+								ObjectRelationshipConstants.TYPE_ONE_TO_MANY) &&
+							 (objectRelationship.getObjectDefinitionId1() ==
+								 _objectDefinition.getObjectDefinitionId()))) {
+
+							_addObjectRelationshipPathItem(
+								key, objectRelationship, paths,
+								relatedSchemaName);
+						}
```

Comment:
Are these changes related to your PR? Can you take a look carefully here? If not, please, remove it.